src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -684,8 +687,54 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
         });
     }
 
+    /**
+     * @description DataImport__c.Donation_Record_Type_Name__c field is mapped from one
+     *              data type to another (Text to Lookup).  This method:
+     *              1. ensures that when the dataImport object is loaded this field always
+     *              holds the RecordType Id value (not the RecordType Name), and
+     *              2. passes the RecordType Id to the setRecordTypeOnFields method so that
+     *              the recordTypeId property is set on sibling fields.
+     * @param dataImport The object being loaded into the form renderer.
+     * @returns {boolean} Returns false if called with a non-null falsy parameter value.
+     */
+    preprocessRecordTypeNameField(dataImport) {
+        const recordTypeNameValue = dataImport[DONATION_RECORD_TYPE_NAME.fieldApiName];
+        if (!recordTypeNameValue && recordTypeNameValue !== null) {
+            return false;
+        }
+
+        // If value is the RecordType Name, change it to the RecordType Id since child
+        // fields are expecting the Id and not the Name.
+        if (recordTypeNameValue &&
+            !recordTypeNameValue.value &&
+            !recordTypeNameValue.startsWith('012')) {
```

Comment:
I'm not sure there's a better way to handle this, but if I get this right you are identifying cases where this field already holds a record type ID (vs Name) by whether it starts with the record type prefix of 012? it seems just a *little* touchy to me, and would fail in at least one extremely silly case (a record type name that starts with "012") ... but like I said not sure if I have a better way to do it. 

If it's not a big lift it might be nice to avoid hardcoding the prefix string here - defining as constant maybe or checking to see if there's an equivalent in LWC to the "findObjNameFromRecordIdPrefix" call from apex?

Reply:
Yep, good call!  I changed it to check if the value is an existing Opportunity RecordType Name instead of the id prefix check.  Let me know what you think of this!